src/candela/components/ScatterPlot/index.js

```diff
@@ -85,6 +85,19 @@ export default class ScatterPlot extends VisComponent {
   }
 
   render () {
-    vega.parseChart(spec, this.el, this.options);
+    this.chart = vega.parseChart(spec, this.el, this.options);
+  }
+
+  getSerializationFormats () {
+    return ['png', 'svg'];
+  }
+
+  serialize (format) {
+    if (!this.chart) {
+      this.render();
```

Comment:
I don't think serializing should cause a render. In this case I'd prefer to return a fulfilled promise that delivers `undefined` or `null` or a blank image.